libs/gltfio/src/ResourceLoader.cpp

```diff
@@ -173,6 +173,63 @@ static void convertBytesToShorts(uint16_t* dst, const uint8_t* src, size_t count
     }
 }
 
+static void decodeDracoMeshes(FFilamentAsset* asset) {
+    DracoCache* dracoCache = &asset->mDracoCache;
+
+    // For a given primitive and attribute, find the corresponding accessor.
+    auto findAccessor = [](const cgltf_primitive* prim, cgltf_attribute_type type, cgltf_int idx) {
+        for (cgltf_size i = 0; i < prim->attributes_count; i++) {
+            const cgltf_attribute& attr = prim->attributes[i];
+            if (attr.type == type && attr.index == idx) {
+                return attr.data;
+            }
+        }
+        return (cgltf_accessor*) nullptr;
+    };
+
+    // Go through every primitive and check if it has a Draco mesh.
+    for (auto pair : asset->mPrimitives) {
+        const cgltf_primitive* prim = pair.first;
+        VertexBuffer* vb = pair.second;
+        if (!prim->has_draco_mesh_compression) {
+            continue;
+        }
+
+        const cgltf_draco_mesh_compression& draco = prim->draco_mesh_compression;
+
+        // Check if we have already decoded this mesh.
+        DracoMesh* mesh = dracoCache->findOrCreateMesh(draco.buffer_view);
+        if (!mesh) {
+            slog.w << "Cannot decompress mesh, Draco decoding error." << io::endl;
+            continue;
+        }
+
+        // Copy over the decompressed data, converting the data type if necessary.
+        if (prim->indices) {
+            mesh->getFaceIndices(prim->indices);
+        }
+
+        // Go through each attribute in the decompressed mesh.
+        for (cgltf_size i = 0; i < draco.attributes_count; i++) {
+
+            // In cgltf, each Draco attribute's data pointer is an attribute id, not an accessor.
+            const uint32_t id = draco.attributes[i].data - asset->mSourceAsset->accessors;
```

Comment:
@romainguy this line was a little tricky, let me know if looks wrong.

Reply:
I actually think the way you did it is reasonable, otherwise you would've had to define a whole new attribute structure just for Draco.  Alternatively I guess you could've modified the existing attribute structure by adding the draco id (perhaps in a union with the accessor pointer) but that seems equally ugly to me.

For now I think we should just take note, and move on.